pandas/core/arrays/sparse/accessor.py

```diff
@@ -227,15 +227,24 @@ def from_spmatrix(cls, data, index=None, columns=None):
         1  0.0  1.0  0.0
         2  0.0  0.0  1.0
         """
-        from pandas import DataFrame
+        from pandas import DataFrame, SparseDtype
+        from . import IntIndex, SparseArray
 
         data = data.tocsc()
```

Comment:
Could be `tocsc(copy=False)` for newer scipy versions, but the gain in performance is minimal anyway with respect to the total runtime.